contracts/CATFactory.sol

```diff
@@ -35,8 +33,8 @@ contract CATFactory is Ownable {
         string memory name,
         string memory symbol
     ) public returns (address) {
-        _tokenIds.increment();
-        uint256 newTokenId = _tokenIds.current();
+        uint256 newTokenId = _nextTokenId;
```

Comment:
We can remove this line, move line 37 after line 51 and use `_nextTokenId` instead of `newTokenId` in line 51.